scripts/notifySlack.js

```diff
@@ -0,0 +1,89 @@
+const fetch = require("node-fetch");
+const repoName = process.env["CIRCLE_PROJECT_REPONAME"];
+const webhookUrl = process.env["SLACK_WEBHOOK_URL"];
+const username =
+  process.env["CIRCLE_PR_USERNAME"] ||
+  process.env["CIRCLE_USERNAME"] ||
+  "default user";
+
+const prUrl = process.env["CIRCLE_PULL_REQUEST"];
+const circleJobUrl = process.env["CIRCLE_BUILD_URL"];
+
+const notifyPass = async (numSuites) => {
+  const msg = `🎉 ✅ Yay! *${username}* passed all ${numSuites} tests in ${repoName}!`;
+  console.log(msg);
+  await postToSlack(webhookUrl, msg);
+};
+
+const notifyFail = async ({ numSuites, passedSuites, failedSuites }) => {
+  const msg = `😿 🚫 *${username}* passed ${
+    passedSuites.length
+  } / ${numSuites} tests for ${repoName}
+Failing tests:
+
+• ${failedSuites.join(" \n • ")}
+  `;
+  console.log(msg);
+  await postToSlack(webhookUrl, msg);
+};
+
+const slackButton = (text, url, style = "primary") => ({
+  type: "button",
+  text: {
+    type: "plain_text",
+    text,
+    emoji: true,
+  },
+  value: text,
+  url,
+  style,
+});
+
+const slackButtons = () => {
+  const buttons = [];
+  if (prUrl) {
+    buttons.push(slackButton("Visit GitHub PR", prUrl));
+  }
+  if (circleJobUrl) {
+    buttons.push(slackButton("See Test Results", circleJobUrl));
+  }
+  return buttons;
+};
+
+const slackPostBody = (msg) => ({
```

Comment:
Slack's API offers a way to send JSON that represents a nicely formatted message with many cool things like interactive buttons. [Lots of examples and an interactive playground here.](https://app.slack.com/block-kit-builder/T02996CND#%7B%22blocks%22:%5B%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22You%20have%20a%20new%20request:%5Cn*%3CfakeLink.toEmployeeProfile.com%7CFred%20Enriquez%20-%20New%20device%20request%3E*%22%7D%7D,%7B%22type%22:%22section%22,%22fields%22:%5B%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Type:*%5CnComputer%20(laptop)%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22*When:*%5CnSubmitted%20Aut%2010%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Last%20Update:*%5CnMar%2010,%202015%20(3%20years,%205%20months)%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Reason:*%5CnAll%20vowel%20keys%20aren't%20working.%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Specs:*%5Cn%5C%22Cheetah%20Pro%2015%5C%22%20-%20Fast,%20really%20fast%5C%22%22%7D%5D%7D,%7B%22type%22:%22actions%22,%22elements%22:%5B%7B%22type%22:%22button%22,%22text%22:%7B%22type%22:%22plain_text%22,%22emoji%22:true,%22text%22:%22Approve%22%7D,%22style%22:%22primary%22,%22value%22:%22click_me_123%22%7D,%7B%22type%22:%22button%22,%22text%22:%7B%22type%22:%22plain_text%22,%22emoji%22:true,%22text%22:%22Deny%22%7D,%22style%22:%22danger%22,%22value%22:%22click_me_123%22%7D%5D%7D%5D%7D)